src/App.tsx

```diff
@@ -1,61 +1,118 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User, TodoWithUser } from './types';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUserById = (userId: number, users: User[]) => (
+  users.find(user => user.id === userId) || null
+);
+
+const getTodoId = (todos: TodoWithUser[]) => {
+  const id = Math.max(...todos.map(todo => todo.id));
+
+  return id + 1;
+};
+
+const todosWithUser = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId, usersFromServer),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<TodoWithUser[]>(todosWithUser);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setUserIdError] = useState(false);
+
+  const addNewTodo = (todoTitle: string, todoUserId: number) => {
+    const newTodo: TodoWithUser = {
+      id: getTodoId(todos),
+      title: todoTitle,
+      completed: false,
+      userId: todoUserId,
+      user: getUserById(todoUserId, usersFromServer),
+    };
+
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    if (!hasUserIdError && !hasTitleError) {
+      addNewTodo(title, userId);
+      setTitle('');
+      setUserId(0);
+    }
+  };
```

Comment:
![Screenshot_88](https://user-images.githubusercontent.com/63102835/199486199-9c817f55-279f-42dd-b402-c2ef008f7e68.jpg)

it is possible to add an empty title with blank space using your code, you need to fix it